src/nft_contract/src/test/mint_to_public.nr

```diff
@@ -0,0 +1,56 @@
+use crate::NFT;
+use crate::test::utils;
+
+#[test]
+unconstrained fn nft_mint_to_public_success() {
+    // Setup without account contracts. We are not using authwits here, so dummy accounts are enough
+    let (env, nft_contract_address, owner, minter, _) =
+        utils::setup_with_minter(/* with_account_contracts */ false);
+    let token_id = 10000;
+
+    env.impersonate(minter);
+    NFT::at(nft_contract_address).mint_to_public(owner, token_id).call(&mut env.public());
+
+    env.advance_block_by(1);
+
+    utils::assert_owns_public_nft(env, nft_contract_address, owner, token_id);
+}
+
+#[test(should_fail_with = "caller is not minter")]
+unconstrained fn nft_mint_to_public_fail_non_minter() {
+    let (env, nft_contract_address, owner, _, recipient) =
+        utils::setup_with_minter(/* with_account_contracts */ false);
```

Comment:
considering we have the LSP that does this, i find the `/* with_account_contracts */` redundant

Reply:
addressing in #116